plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/OpenShiftImages.java

```diff
@@ -30,4 +33,21 @@
 	public static final Image GEAR_IMG = repo.getImage("gear.png"); //$NON-NLS-1$ 
 	public static final ImageDescriptor LAYER = repo.create("layer.png"); //$NON-NLS-1$ 
 	public static final Image LAYER_IMG = repo.getImage("layer.png"); //$NON-NLS-1$ 
+	
+	public static Image getImage(IResource resource) {
```

Comment:
The default pattern (and aim) for image classes (OpenShiftImages) is to have them as simple repository/cache for images since the problem here is that images are native resources which have to be manually disposed.
Adding logic here that's returning an image for a given resource is what you'd usually put into a label provider or a utils class (if the pattern recurrent). Please move this logic to the appropriate label provider.